src/app/components/NetworkBadge/NetworkBadge.tsx

```diff
@@ -1,21 +1,23 @@
 import Image from "next/image";
+import { twJoin } from "tailwind-merge";
 
+import { useWallet } from "@/app/context/wallet/WalletProvider";
 import { network } from "@/config/network.config";
 import { Network } from "@/utils/wallet/wallet_provider";
 
 import testnetIcon from "./testnet-icon.png";
 
-// This component can also be used rendering based on the network type
-interface NetworkBadgeProps {
-  isWalletConnected: boolean;
-}
+export const NetworkBadge = () => {
+  const { walletProvider } = useWallet();
 
-export const NetworkBadge: React.FC<NetworkBadgeProps> = ({
-  isWalletConnected,
-}) => {
   return (
     <div
-      className={`absolute left-2 ${isWalletConnected ? "top-40 md:top-24 lg:top-32" : "top-24 md:top-24 lg:top-32"}`}
+      className={twJoin(
+        `absolute left-2`,
+        Boolean(walletProvider)
```

Comment:
`Boolean` vs `!!`? Later on I suggest sticking to one approach, right now we have 8 `!!` and 6 `Boolean(something)`